vsintegration/src/FSharp.Editor/Telemetry/TelemetryReporter.fs

```diff
@@ -89,29 +95,47 @@ type TelemetryReporter private (name: string, props: (string * obj) array, stopw
 
     static member val private SendAdditionalTelemetry =
         lazy
-            (let componentModel =
-                Package.GetGlobalService(typeof<ComponentModelHost.SComponentModel>) :?> ComponentModelHost.IComponentModel
-
-             if componentModel = null then
-                 TelemetryService.DefaultSession.IsUserMicrosoftInternal
-             else
-                 let workspace = componentModel.GetService<VisualStudioWorkspace>()
-                 workspace.Services.GetService<EditorOptions>().Advanced.SendAdditionalTelemetry)
+            (
+                let componentModel =
+                    Package.GetGlobalService(typeof<ComponentModelHost.SComponentModel>) :?> ComponentModelHost.IComponentModel
+
+                if isNull componentModel then
+                    TelemetryService.DefaultSession.IsUserMicrosoftInternal
+                else
+                    let workspace = componentModel.GetService<VisualStudioWorkspace>()
+
+                    TelemetryService.DefaultSession.IsUserMicrosoftInternal 
+                    || workspace.Services.GetService<EditorOptions>().Advanced.SendAdditionalTelemetry)
+
+    static member ReportFault(name, ?severity: FaultSeverity, ?e: exn) =
+        if TelemetryReporter.SendAdditionalTelemetry.Value then
+            let faultName = String.Concat(name, "/fault")
```

Comment:
I guess ideally the concatenation should be smarter, as in - it shouldn't add a slash if there already is one.

That's to be future-proof since one might call this with `my/event/` and reviewers might forget that the function assumes no slash.

Reply:
We don't really want to do any additional logic here at this point, since it's quite a hot path. We might do something around events themselves, to be more strongly typed.

Reply:
Makes sense, created a ticket for that.